frb_rust/src/rust2dart.rs

```diff
@@ -139,3 +145,120 @@ impl<T: IntoDart> StreamSink<T> {
         self.rust2dart().close_stream()
     }
 }
+
+/// Basically the Into trait.
+/// We need this separate trait because we need to implement int for Vec<T> etc.
+pub trait IntoIntoDart<D> {
+    fn into_into_dart(self) -> D;
+}
+
+impl<T, D> IntoIntoDart<Vec<D>> for Vec<T>
+where
+    T: IntoIntoDart<D>,
+{
+    fn into_into_dart(mut self) -> Vec<D> {
+        self.drain(0..).map(|e| e.into_into_dart()).collect()
```

Comment:
nit: does this work?

```suggestion
        self.into_iter().map(|e| e.into_into_dart()).collect()
```

Reply:
yes it does. I will change that